src/openssl.cpp

```diff
@@ -135,6 +164,9 @@ int openssl_create_csr(const lmp_options &options, string &pkey, string &csr)
 	if (add_ext(ext, NID_ext_key_usage, (char *)"clientAuth"))
 		leave;
 
+	if (add_custom_ext(ext, CUSTOM_EXT_MPROTECT_KEY_OID, opt.mprotect_key))
```

Comment:
I dont remember having made any changes to the backend. ever. 

if there is nobody to look into the integration of the feature I can have a look over the coming weeks (probably after the Security Summit end of September). But as you say I do'nt expect any server to be able to work auto-magically with a custom extension.

if it is too painful I can go back to pass it differently.

Reply:
I'm not trying to dictate who changes the backend. The PR just didn't imply that its really an untested work in progress. I suspect the other reviewers may not have understood this.

Reply:
I doubt (99%) there will be any issues signing the CSR - it just that the key will probably be removed since it is a custom oid.

I had envisioned the CSR to be parsed and the key retrieved if present. Then continue with business as usual - plus storing the key somewhere.

but yeah, this has had no end to end testing. just local to generate the CSR after reading the key from the CAAM hardware via the TEE client.

Reply:
IIUC the mprotect key is simply the `prime256v1` key.

If so, why can't we use it as a key for the device client certificate, and sign it via CAAM?
Then ak-lite would be amended to do the TLS-time encryption through the CAAM module as well.

This way, no changes are needed to the back-end at all, as CAAM is incorporated transparently on the device side.

Going further, the customers might extend their _registration server_ to validate that the received client certificate:
- has a unique pubkey;
- has a pubkey which belongs to a predefined set;
- has a uuid which belongs to a predefined set;
- whatever else they want to verify.

Reply:
We can't use this key with pkcs11 at this moment, and not every device would necessarily want the CAAM based key to be used for TLS (e.g. with SE050). Which is why I think it is just better to handle it as a separated key.